### PR TITLE
Added category entity and all related changes.

### PR DESCRIPTION
**Pull request by k0sm0naft**

### CODE REVIEW

File: src/main/java/ua/bookstore/online/controller/BookController.java [hunk 1]
```diff
@@ -101,14 +102,14 @@ public BookDto getBookById(@PathVariable @Parameter(description = "Book ID") Lon
             @ApiResponse(responseCode = "401", description = "Required authorization",
                     content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
     })
-    public List<BookDto> getAll(
+    public List<BookDtoWithoutCategoryIds> getAll(
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/controller/BookController.java]: why without categories? I think you'd like to see categories when you want to get all books

File: src/main/java/ua/bookstore/online/controller/CategoryController.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Successfully created"),
+            @ApiResponse(responseCode = "400", description = "Invalid request body",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
+            @ApiResponse(responseCode = "401", description = "Required authorization",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
+            @ApiResponse(responseCode = "403", description = "Not enough access rights",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class)))
+    })
+    public CategoryResponseDto createCategory(@RequestBody CategoryRequestDto categoryDto) {
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/controller/CategoryController.java]: add `@Valid`

File: src/main/java/ua/bookstore/online/controller/CategoryController.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "202", description = "Successfully updated"),
+            @ApiResponse(responseCode = "400", description = "Invalid request body",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
+            @ApiResponse(responseCode = "401", description = "Required authorization",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
+            @ApiResponse(responseCode = "403", description = "Not enough access rights",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class)))
+    })
+    public CategoryResponseDto updateCategory(@PathVariable Long id, @RequestBody
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/controller/CategoryController.java]: add `@Valid`

File: src/main/java/ua/bookstore/online/controller/CategoryController.java [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    @GetMapping
+    @PreAuthorize("hasRole('USER')")
+    @ResponseStatus(HttpStatus.OK)
+    @Operation(summary = "Return all of categories", description = "Return all of categories")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successfully retrieved"),
+            @ApiResponse(responseCode = "401", description = "Required authorization",
+                    content = @Content(schema = @Schema(implementation = ErrorResponseDto.class))),
+    })
+    public List<CategoryResponseDto> getAll() {
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/controller/CategoryController.java]:
```suggestion
    public List<CategoryResponseDto> getAll(Pageable pageable) {
```

File: src/main/java/ua/bookstore/online/dto/category/CategoryRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package ua.bookstore.online.dto.category;
+
+import jakarta.validation.constraints.NotBlank;
+
+public record CategoryRequestDto(
+        @NotBlank
+        String name,
+        @NotBlank
+        String description
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/dto/category/CategoryRequestDto.java]: let's add size constraint

File: src/main/java/ua/bookstore/online/model/Category.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@Entity
+@Table(name = "categories")
+public class Category {
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/model/Category.java:18]: I think you'd like to have soft delete here, won't you?

File: src/main/java/ua/bookstore/online/service/BookService.java [hunk 7]
```diff
@@ -3,19 +3,23 @@
...
 import ua.bookstore.online.dto.book.BookDto;
+import ua.bookstore.online.dto.book.BookDtoWithoutCategoryIds;
 import ua.bookstore.online.dto.book.CreateBookRequestDto;
 import ua.bookstore.online.dto.search.parameters.BookSearchParameters;
 
 public interface BookService {
     BookDto save(CreateBookRequestDto bookRequestDto);
 
-    List<BookDto> getAll(Pageable pageable);
+    List<BookDtoWithoutCategoryIds> getAll(Pageable pageable);
```
Comment (yevheniisynytsia) [inline src/main/java/ua/bookstore/online/service/BookService.java]: same here, why without categories?

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Great job! Left some comments

File: src/main/java/ua/bookstore/online/controller/CategoryController.java (hunk 4 above)
Comment (k0sm0naft) [reply src/main/java/ua/bookstore/online/controller/CategoryController.java]: Not sure that here it is needed, but it is not a bother.

File: src/main/java/ua/bookstore/online/model/Category.java (hunk 6 above)
Comment (k0sm0naft) [reply src/main/java/ua/bookstore/online/model/Category.java:18]: No. I think it is not necessary here. (Discussed)

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**